osu.Game/Overlays/OnlineOverlay.cs

```diff
@@ -65,10 +67,21 @@ protected OnlineOverlay(OverlayColourScheme colourScheme, bool requiresSignIn =
                         },
                     }
                 },
-                Loading = new LoadingLayer(true)
+                loadingContainer = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Child = Loading = new LoadingLayer(true),
+                }
             });
 
             base.Content.Add(mainContent);
         }
+
+        protected override void UpdateAfterChildren()
+        {
+            base.UpdateAfterChildren();
+
+            loadingContainer.Padding = new MarginPadding { Top = Math.Max(0, Header.Height - ScrollFlow.Current) };
```

Comment:
I'd want to see an inline comment explaining what this is doing, since it's a bit weird logic (and is run every `Update` loop).